components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/ClientManager.java

```diff
@@ -93,9 +94,19 @@ public CloseableHttpAsyncClient getClient(String tenantDomain) throws FrameworkE
     private RequestConfig createRequestConfig() {
 
         return RequestConfig.custom()
-                .setConnectTimeout(HTTP_CONNECTION_TIMEOUT)
-                .setConnectionRequestTimeout(HTTP_CONNECTION_REQUEST_TIMEOUT)
-                .setSocketTimeout(httpReadTimeout)
+                .setConnectTimeout(StringUtils.isNotBlank(
+                        IdentityUtil.getProperty(Constants.CALL_CHOREO_HTTP_CONNECTION_TIMEOUT)) ?
```

Comment:
if there are any other adaptive functions in the runtime that calls this method to create a request config might get the timeouts you have configured for call choreo function.